libres/lib/res_util/res_env.cpp

```diff
@@ -87,22 +87,20 @@ char *res_env_alloc_PATH_executable(const char *executable) {
         auto path_list = res_env_alloc_PATH_list();
         int ipath = 0;
 
-        while (true) {
-            if (!path_list[ipath].empty()) {
-                char *current_attempt = util_alloc_filename(
-                    path_list[ipath].c_str(), executable, NULL);
-
-                if (util_is_file(current_attempt) &&
-                    util_is_executable(current_attempt)) {
-                    full_path = current_attempt;
-                    break;
-                } else {
-                    free(current_attempt);
-                    ipath++;
-                }
-            } else
+        for (std::size_t i = 0; i < path_list.size(); ++i) {
```

Comment:
How about
```cpp
for (auto path : path_list) {
    char *current_attempt =
        util_alloc_filename(path.c_str(), executable, NULL);
    if (util_is_file(current_attempt) &&
        util_is_executable(current_attempt)) {
        full_path = current_attempt;
        break;
    } else {
        free(current_attempt);
    }
}
return full_path;
```
or
```cpp
auto found_file = std::find_if(path_list.start(), path_list.end(), [&executable](path) {
    char *current_attempt = 
        util_alloc_filename(path_list[ipath].c_str(), executable, NULL);
    return (util_is_file(current_attempt) &&
            util_is_executable(current_attempt));
  });

if (found_file ==path_list.end()) //didn't find any
    return NULL;
else
    return util_alloc_filename(*found_file, executable, NULL);
```

Reply:
For reference, the for-each loop with `std::filesystem`:
```cpp
char *full_path = NULL;
for (auto path : path_list) {
    auto current_attempt = fs::path(path) / executable;
    if(fs::exists(current_attempt) && (current_attempt.permissions() & fs::perms::owner_exec)) {
        full_path = util_alloc_filename(path.c_str(), executable, NULL);
        break;
    }
}
return full_path;
```

Reply:
Thanks, I've pushed a version that uses `for (auto path: path_list)`.
While I do agree that using `filesystem` is much better, the point of this PR was to get rid of `util_split_string` and `util_alloc_joined_string`.

Reply:
No problem! The filesystem part of my comment is being worked on by @pinkwah here: https://github.com/equinor/ert/pull/3517